internal/hcops/load_balancer.go

```diff
@@ -845,6 +845,17 @@ func (l *LoadBalancerOps) ReconcileHCLBServices(
 		return false, fmt.Errorf("%s: %v", op, err)
 	}
 
+	for _, port := range svc.Spec.Ports {
+		if port.Protocol != "" && port.Protocol != corev1.ProtocolTCP {
+			err := fmt.Errorf(
+				"configured unsupported Hetzner Cloud load balancer protocol %s for service with name %s",
+				port.Protocol,
+				svc.Name,
+			)
+			return false, err
+		}
```

Comment:
If we don't plan to fail, shouldn't we continue processing the function, instead of returning? Feels like a breaking change.

Reply:
I always thought we didn't want to break user land for this issue. So I expect this to apply as much as possible and only trigger warning events where we can't/don't support the feature.

But maybe you already discussed a different plan with @apricote?

Reply:
No you are totally right. It would be a breaking change. I missed that and just thought about how I would like to have it :sweat_smile:

Will adapt the code accordingly and dispatch an event/log msg here instead of failing.

Reply:
Yea, I would also prefer it if we could fail completely, but that would be a breaking change.